beacon_node/network/src/sync/network_context.rs

```diff
@@ -358,6 +371,7 @@ impl<T: BeaconChainTypes> SyncNetworkContext<T> {
         batch_type: ByRangeRequestType,
         request: BlocksByRangeRequest,
         sender_id: RangeRequestId,
+        syncing_peers: impl Iterator<Item = PeerId>,
```

Comment:
This is not syncing peers, but the set of peers that claim to be able to serve `request`. Why take an iterator as arg instead of just `Vec<PeerId>` or a slice?

Reply:
The name came from here (peers of a `SyncingChain`) - but can rename if it's misleading
https://github.com/sigp/lighthouse/blob/614f98452fb3eda2d910f956d2ef7190530be318/beacon_node/network/src/sync/range_sync/chain.rs#L178-L181